api/types/plugin_static_credentials.go

```diff
@@ -0,0 +1,141 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package types
+
+import "github.com/gravitational/trace"
+
+// PluginStaticCredentials are static credentials for plugins.
+type PluginStaticCredentials interface {
+	// ResourceWithSecrets provides common resource methods.
+	ResourceWithLabels
+
+	// GetAPIToken will return a flag indicating whether this object contains an API token
+	// credentials and the attached API token if possible.
+	GetAPIToken() (bool, string)
+
+	// GetAPIToken will return a flag indicating whether this object contains basic auth
+	// credentials and the attached username and password if possible.
+	GetBasicAuth() (bool, string, string)
+
+	// GetOAuthClientSecret will return a flag indicating whether this object contains OAuth
+	// client secret credentials and the attached client ID and client secret.
+	GetOAuthClientSecret() (bool, string, string)
```

Comment:
Two comments on these:

- Why do they need the first "bool" return argument? I understand it indicates whether this particular type of credential is set or not, but can't we just determine this by whether the credential itself is set or not?
- I'd use named return arguments here for clarity.

Reply:
The `bool` argument isn't necessary, but IMO makes it a little more friendly to work with, i.e. no having to compare various fields to `nil` in order to determine if they're present.

I don't feel strongly about this, so if you have a strong feeling about how it should be, I'm fine with that.

Reply:
No strong feelings, just wondering how useful it will be in practice - i.e. why would the caller need to look at the boolean argument instead of just seeing whether credentials themselves are set or not?

Reply:
Ahhh, I see what you're getting at. Yeah, that makes sense. I'll make that change.